### DISCUSSION TITLE
Installation advices for fajita

### DISCUSSION THREAD

**Discussion by edern-appere** [Divested-Mobile/DivestOS-Build#108] [General] [answered]
Hi,

I've read the installation procedure described on this page : https://divestos.org/index.php?page=bootloader
I have some questions before I try to make the switch (I don't want to brick my phone !) :
- my phone is already unlocked. Should I do the first 4 steps in the installation tutorial ?
- my phone is currently running LOS+microg. Do i have to wipe everything in order to install DivestOs ?
- How do i know if my phone is A/B or AVB ?
- Locking the bootloader seems a good security practice but should I wait a little bit (testing everything) before i do it ?
Thanks for your help !

**Comment by SkewedZeppelin** [comment, marked as answer]
Start at step 5. Follow 5, 7, 9, 10, 11, 12.

fajita is both A/B and AVB, you can see this by the fact that copy-partitions.zip and avb.bin is available on the download page.
firmware is included on fajita/enchillada so you can skip copy-partitions (step 8).

A wipe is strongly recommended, but you can try to see if it boots without.
Otherwise do wipe if it doesn't boot after a minute or two.

I've personally tested fajita working locked.

**Reply by edern-appere** [reply]
Thanks a lot ! 
I hope to find time to install DivestOS one of these days :-)

**Reply by edern-appere** [reply]
I can wipe, granted seedvault is installed in DivestOS and I can recover my backups with it (would be a first time for me...)

**Reply by SkewedZeppelin** [reply]
**Do not rely** on SeedVault, it cannot backup all apps, doesn't backup files, and it has issues with corruption sometimes.

Use the built-in backup functions in all available apps first.
And also manually copy all of your internal storage to a computer first (pictures/downloads/documents).

**Comment by edern-appere** [comment]
I always backup my apps manually first and the main folders containing files, but some apps do not have backup/export features and for security reasons, one cannot access android data folder. It's getting more and more complicated !

**Comment by d4rklynk** [comment]
To earse avb key and install the DOS one, is it from the bootloader of from the DOS recovery ?

**Reply by SkewedZeppelin** [reply]
Bootloader as it is a fastboot command.

**Reply by d4rklynk** [reply]
Done ! Everything is working well as expected

**Reply by edern-appere** [reply]
@d4rklynk : did you install on Fajita ? Is everything working fine ?

**Reply by d4rklynk** [reply]
Everything working fine except location (GPS) that I'm trying to make it work. ~My bank doesn't work~, Notifications don't work except some apps like Briar, Signal, and Element.
If you use it before, Netflix and Disney+ don't work.
Spotify works ! :heart: 
But I have bootloader locked and that is better tbh. I'm gonna try a few days, before really decide if I keep it or not.
Fingerprint doesn't work well, but mostly because the sensor is really bad (so thanks OnePlus for this one).

Don't be confused, if you read to fast, step 11 **doesn't lock** the bootloader, it is just installing the signature of the ROM.
After that step 11 you boot and try to check if everything works, updater works, and do a factory reset from the settings of Android.
After that, you can lock the bootloader if everything works as expected.
The ROM itself works really well, love it.

**Reply by d4rklynk** [reply]
For the records, I "Factory Reset" from LineageOS **recovery** before starting all the steps
I still did it again at step 10, in case.

**Reply by edern-appere** [reply]
Thanks for the tips ! Apart from a lack of time the next few days, I'd realy like to install it also and your tips are great. The only thing that annoys me is the lack of location... 
I often use OSMAnd and Here Maps but I guess they will work anyway. But some apps that rely on MicroG / UnifiedNlp, like Transportr or my car-sharing app, will they work ? This is really a deal-braker for me :-/

**Reply by d4rklynk** [reply]
Google Maps work, but the location won't because I think it needs at least MicroG for that.
But here location doesn't work at all, I don't know how to do.
I just love Google Maps, it works really great. But I wanted to use at least GMaps Webview Scrapper on F-droid (use DivestOS repo) (iirc the name) to have maps. And use OsmAnd+ for driving with GPS. But for now GPS doesn't work.

**Reply by d4rklynk** [reply]
**UPDATE :** My bank actually works, I don't think I'll change ROM then x)

**Reply by d4rklynk** [reply]
Note : If you want a play store I suggest you to use Aurora Store (in "Session mode"). 
When you will install an app : it will prompt you.
Now with android 12, it willl still prompt you when you install something, but **not** when you will update it ! Really neat !

Same goes for **Droidi-fy** (F-Droid Client) aka Neo-store https://github.com/NeoApplications/Neo-Store.
Really convenient.

**Reply by d4rklynk** [reply]
**Important :** You need to be sure that you are on the latest firmware (the very last one is 11.1.2.2), because you gonna lock the BL (bootloader).
If something is not working like camera, or Wifi sensors, you would have to unlock the BL to be able to flash the firmware.
Link to the latest fimrware of fajita : https://otafsg1.h2os.com/patch/amazone2/GLO/OnePlus6Oxygen/OnePlus6Oxygen_22.J.62_GLO_0620_2111252336/OnePlus6Oxygen_22.J.62_OTA_0620_all_2111252336_14afec75dd6fa.zip

I had this link from this website https://www.getdroidtips.com/oneplus-6t-stock-firmware-collections/ (if you don't trust my link, bc you shouldn't x))

Link to LineageOS to install firmware : https://wiki.lineageos.org/devices/fajita/fw_update

**Reply by SkewedZeppelin** [reply]
Firmware is already included: https://gitlab.com/divested-mobile/firmware-empty/-/blob/master/STATUS

GPS does work! microG/UnifiedNlp is NOT required for location to apps.
Per the website:
> Most devices will take longer to acquire a GPS lock. Please test using GPSTest app with *clear* and *direct* line of sky. [deblobber][[link]](https://f-droid.org/en/packages/com.android.gpstest.osmdroid)

A full GPS lock can take up to 12.5 minutes.

And I do NOT recommend Droid-ify!

If you use apps from Aurora, I suggest you put them in a work profile via Shelter.

**Reply by SkewedZeppelin** [reply]
GMaps WV is also not a scraper, it connects directly to Google.
And it is available, like most of my apps, on the official F-Droid repo.

**Reply by d4rklynk** [reply]
> Firmware is already included: https://gitlab.com/divested-mobile/firmware-empty/-/blob/master/STATUS

Nice to hear !
> GPS does work! microG/UnifiedNlp is NOT required for location to apps. Per the website:
> > Most devices will take longer to acquire a GPS lock. Please test using GPSTest app with _clear_ and _direct_ line of sky. [deblobber][[link]](https://f-droid.org/en/packages/com.android.gpstest.osmdroid)

I'm gonna try that ! I guess it is not really accurate bc it uses GPS only right ? And does it gonna take always a dozen minutes every time or is it just a one time thing ?

> And I do NOT recommend Droid-ify!

You talked about that on this discussion right ? https://github.com/Divested-Mobile/DivestOS-Build/discussions/58

> If you use apps from Aurora, I suggest you put them in a work profile via Shelter.

Even for apps like Signal ?

> GMaps WV is also not a scraper, it connects directly to Google.

Good to know actually

**Reply by SkewedZeppelin** [reply]
> I guess it is not really accurate bc it uses GPS only right 

GPS is the most accurate location provider there is.

> does it gonna take always a dozen minutes

It is only *up to* 12.5 minutes, due to how the satellites are physically moving above you in the sky.
Depending on your/their position it can lock quicker.
And usually subsequent uses are quicker.
The last location will also be cached by the system.

> Signal

less egregious, but still includes proprietary blobs, fine to keep in main profile.

**Reply by d4rklynk** [reply]
Yeah GPS totally works and really accurate , just had to go outside.

**Reply by d4rklynk** [reply]
Thanks a **looot** mate, DivestOS is really cool and you are really helpful. Thanks for all that !

**Reply by edern-appere** [reply]
> Yeah GPS totally works and really accurate , just had to go outside.

So I guess it took you less than 12 min ?

**Reply by d4rklynk** [reply]
Yeah I open my door, I went outside, done. 1 second.

**Reply by edern-appere** [reply]
I don't know why but I thought microG/UnifiedNlp was here also to provide maps to apps requiring GoogleMaps (like Uber) ?

**Reply by d4rklynk** [reply]
No even Google Maps works fine.

**Reply by d4rklynk** [reply]
Aah, i don't know x)

**Reply by edern-appere** [reply]
> **UPDATE :** My bank actually works, I don't think I'll change ROM then x)

Did you make any tweaks so that it works or not ? My bank app does not work :-(

**Comment by edern-appere** [comment]
I have started making my manual backups (Dcim, Downloads and so on) and I have additional questions :
- will I be prompted to restore seedvault backup on first boot (useful for some apps I cannot backup manualy like Whatsapp) ?
- if I lock the bootloader, will I be able to unlock it after, if I want to switch to another rom (for example) ?
Have a nice day !

**Reply by SkewedZeppelin** [reply]
> prompted to restore seedvault backup

yes, but it is clunky, you have to pull down the notifications, enable file transfer, and copy the backup in place.

> will I be able to unlock it after

of course, just be sure to keep oem unlocking checked in the developer settings. you can disable the dev settings and it'll stick

**Reply by edern-appere** [reply]
Yeah I do know seedvault is clunky. Last time I had to reinstall my phone, I had seedvault backups on Nextcloud but the app never asked me to get it from this place... I tried putting the files manualy inside seedvault folder, to no avail... I still don't get how seedvault works yet I still hope it will work everytime I have to backup my phone.

**Comment by edern-appere** [comment]
I finaly had the time to install DivestOs. 

I tried to do the steps you recommended : "Start at step 5. Follow 5, 7, 9, 10, 11, 12." but there were some error messages... I tried to install nonetheless. Here are the messages :

`
sudo fastboot update divested-19.1-20220511-dos-fajita-fastboot.zip
extracting android-info.txt (0 MB)...
extracting boot.img (64 MB)...
target reported max download size of 268435456 bytes
archive does not contain 'boot.sig'
archive does not contain 'boot_other.img'
extracting dtbo.img (8 MB)...
archive does not contain 'dtbo.sig'
archive does not contain 'dt.img'
archive does not contain 'recovery.img'
extracting system.img (1588 MB)...
archive does not contain 'system.sig'
archive does not contain 'system_other.img'
extracting vbmeta.img (0 MB)...
archive does not contain 'vbmeta.sig'
extracting vendor.img (578 MB)...
W/ziparchive( 8011): Zip: unable to allocate  606392648 bytes at offset 0 : No space left on device
failed to extract 'vendor.img': I/O error
archive does not contain 'vendor_other.img'
--------------------------------------------
Bootloader Version...: unknown
Baseband Version.....: 
Serial Number........: XXXXXXX
--------------------------------------------
checking product...
FAILED

Device product is 'OnePlus6T'.
Update requires 'sdm845'.

finished. total time: 0.002s
`

I followed the next steps and now the OS is on the phone. It booted normaly... Now it runs ok and is crazy fast, like if I bought a new phone ! I haven't tried everything yet (like bluetooth and so on).
Seedvault restore (from nextcloud) almost worked !

**Reply by SkewedZeppelin** [reply]
> unable to allocate

It looks like you ran out of space in /tmp on your computer.

As long as you did the sideload via recovery after, it should be installed correctly.

**Reply by edern-appere** [reply]
That must be the case, my computer is running out of space (at least my OS partition). 
I did the sideload via recovery. I thought it would do less harm to make the jump entirely than stop in the middle of the process. 
So far, everything works fine. I hope the lack of space on my laptop hasn't hindered some of the update processes.

**Reply by edern-appere** [reply]
BTW, thanks a lot for making DivestOS and all your usefull tips. It helped a lot !

**Comment by edern-appere** [comment]
I've noticed some pings to connectivitycheck.android.com. I've read this could be disabled in the settings (https://divestos.org/index.php?page=network_connections) but I could not get the hand on this... Any idea ?

**Reply by SkewedZeppelin** [reply]
That feature has yet to be updated for 19.1, https://divestos.org/index.php?page=news-future#twelve

You can disable it via ADB via `adb shell settings put global captive_portal_mode 0;`, https://divestos.org/index.php?page=privacy_policy

**Comment by edern-appere** [comment]
I may be unable to continue using DivestOS. In fact my bank app does not work. This is a really bad news because everytime I pay with my credit card, I have to validate the paiment with my bank's app :-( It doesn't even start... 
I may be forced to switch back to my previous LOS+microg built where the bank app was working. This issue is strange because I didn't even have any Google account registered inside microg.
I tried disabling DNS host filtering and secure app spawning but to no avail...
:-(

**Reply by SkewedZeppelin** [reply]
Have fun!

**Reply by SkewedZeppelin** [reply]
You might double check to see what it depends on if there is a report here: https://akc3n.org/projects/banking/

**Reply by edern-appere** [reply]
> Have fun!

Well, having to make a fresh install once again, set up everything and transfer files is not quite fun. I would prefer to keep DivestOs...

**Reply by SkewedZeppelin** [reply]
It is out of my hands what choices your bank makes.

**Reply by d4rklynk** [reply]
If my banks work fine on DOS, it will definitely work on GrapheneOS right ?

**Reply by edern-appere** [reply]
> It is out of my hands what choices your bank makes.

I know you have nothing to do with this, I won't blame you for my bank's stupid tech decisions !

**Reply by SkewedZeppelin** [reply]
@edern-appere 
Also in case you have it disabled, enable "Enable native code debugging" and try again.

@d4rklynk 
Very likely.

**Reply by edern-appere** [reply]
No luck !

**Comment by edern-appere** [comment]
Hi there,
I finally had to go back to LineageOs+microG rom... which makes it possible for my bank app to work. I'm sorry to have bothered you for the last few days with all my questions about the installation process and so on. I would like to thank you for your help, your time, and all the great work you put into DivestOs. I'd recommend this Os to anyone looking for a secure Os.
I hope this thread will help some people installing DivestOs on their Fajita :-)
Have a nice day !
